docs/extended_api/synchronization_library/pipeline_consumer_wait_prior.md

```diff
@@ -0,0 +1,52 @@
+---
+grand_parent: Extended API
+parent: Synchronization library
+---
+
+# cuda::**pipeline_consumer_wait_prior**
+
+Defined in header [`<cuda/pipeline>`](../headers/pipeline.md)
+
+```c++
+template<uint8_t Prior>
+void pipeline_consumer_wait_prior(pipeline<thread_scope_thread> & pipeline);
+```
+
+Blocks the current thread until all operations committed to _pipelines stages_ sequenced before the `Prior` last one complete. All stages up to `Prior` (excluded)
+are implicitly released.
+
+## Template parameters
+
+| Prior | The Nth latest stage to wait for |

```

Comment:
```suggestion
| Prior | The index of the pipeline stage *Stage* (_see above_) counting up from the current one |
```

I admit that this is a bit wonky, but I think it's an improvement - it doesn't introduce `N` with no clear definition of what it is. We may need to also say "the index of the current stage is 0" or something here.

(TBQH I really hate this functionality, only because of how hard it is to word _what it does_ in English.)